src/Umbraco.Web/Editors/CodeFileController.cs

```diff
@@ -0,0 +1,155 @@
+﻿using AutoMapper;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web.Http;
+using Umbraco.Core.Models;
+using Umbraco.Web.Models.ContentEditing;
+using Umbraco.Web.Mvc;
+using Umbraco.Web.WebApi;
+
+namespace Umbraco.Web.Editors
+{
+    [PluginController("UmbracoApi")]
+    public class CodeFileController : UmbracoAuthorizedJsonController
+    {
+
+        public HttpResponseMessage PostCreate(string type, CodeFileDisplay display)
+        {
+            switch (type)
+            {
+                case "partialView":
+                    var view = new PartialView(display.VirtualPath);
+                    var result = Services.FileService.CreatePartialView(view, display.Snippet, Security.CurrentUser.Id); 
+                    return result.Success == true ? Request.CreateResponse(HttpStatusCode.OK, result.Result) : Request.CreateNotificationValidationErrorResponse(result.Exception.Message);
```

Comment:
This is not what should be returned. Since you seem to be wanting to return some data, this action should return that class, not a `HttpResponseMessage`, however you are actually returning a business logic class of `IPartialView` which should never be done since we never expose business logic classes to the angular app, only view models. If you wish to return a view model result, then the result of this action should be `CodeFileDisplay` (assuming that is what you want to return)